api/app/Services/ImageUploader.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Services;
+
+use App\Contracts\FileUploader;
+use Illuminate\Config\Repository;
+use Illuminate\Filesystem\FilesystemManager;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Arr;
+
+final class ImageUploader implements FileUploader
+{
+    private FilesystemManager $fileSystemManager;
+    private Repository $config;
+
+    public function __construct(
+        FilesystemManager $filesystemManager,
+        Repository $config
+    ) {
+        $this->fileSystemManager = $filesystemManager;
+        $this->config = $config;
+    }
+
+    public function upload(UploadedFile $file, int $userId, string $type): string
```

Comment:
The method [should be much simpler](https://laravel.com/docs/7.x/filesystem#file-uploads). Probably we will not need a separate service for it.
Example:
```
        $newFileName = <generate random name>;
        $fileSystemManager->disk()->putFileAs(<files dir from config>, $file, $newFileName, 'public');
        $link = $fileSystemManager->disk()->url(<files dir from config> . $newFileName);
```